### PR TITLE
feat: add stringToIndex util function for SmartAccount salt

### PR DESCRIPTION
**Pull request by DoubleOTheven**
# Pull Request Checklist

- [x] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [x] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [x] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [x] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [x] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

### Detailed summary
- Added a new utility function `stringToIndex` in `aa-core/utils` package.
- Updated the `config.ts` file in the `site/.vitepress` directory to include a new link to `stringToIndex` in the documentation.
- Updated the `utils.test.ts` file in the `packages/core/src/__tests__` directory to include a test for `stringToIndex`.
- Updated the `constructor.md` file in the `site/packages/aa-core/accounts` directory to include a reference to `stringToIndex` and its usage.
- Added a new markdown file `stringToIndex.md` in the `site/packages/aa-core/utils` directory to provide detailed documentation for `stringToIndex` function.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/core/src/utils/bigint.ts [hunk 1]
```diff
@@ -76,3 +77,22 @@ export const bigIntClamp = (
...
+ *   index: salt,
+ *   // other args omitted...
+ * });
+ * ```
+ *
+ * @param phrase -- any string value.
+ * @returns the bigint value of the hashed string
+ */
+export const stringToIndex = (phrase: string): bigint =>
+  BigInt(hashMessage(phrase));
```
Comment (moldy530) [inline packages/core/src/utils/bigint.ts:98]: nit: we could simplify this a bit more to just be `keccak256(toHex(phrase))`

`hashMessage` adds the `0xEthereum Signed Message (len)` prefix to the bytes representation of the `phrase`. Not a big deal tbh, but reads a bit more clearly 🤷‍♂️

**Review by moldy530 — COMMENTED** [review]
This is a great idea! lemme know what you think about the comment and then can approve after that

File: packages/core/src/utils/bigint.ts (hunk 1 above)
Comment (DoubleOTheven) [reply packages/core/src/utils/bigint.ts:98]: Yeah, good call. I'll make that fix now

**Comment by DoubleOTheven** [comment]
@moldy530 Fixed. Thanks for the fast reply.

**Review by moldy530 — APPROVED** [review]

**Pull request merged**